src/aero_kernel/src/utils/mod.rs

```diff
@@ -27,6 +27,22 @@ pub mod io;
 pub mod linker;
 pub mod sync;
 
+pub fn validate_ptr<T>(ptr: *const T) -> Option<&'static T> {
+    if ptr.is_null() {
+        None
+    } else {
+        unsafe { Some(&*ptr) }
+    }
+}
+
+pub fn validate_mut_ptr<T>(ptr: *mut T) -> Option<&'static mut T> {
```

Comment:
`VirtAddr::new(ptr as usize).read_mut::<T>()`

Reply:
fixed in https://github.com/Andy-Python-Programmer/aero/pull/65/commits/82de6408d3bf7b73374475c52d4dd5fa2ec13014